grammar.js

```diff
@@ -24,497 +39,1236 @@ module.exports = grammar({
+
+    _pow_operator: $ => token(choice(
+      seq('**', repeat(OP_CHAR)),
+      'lsl', 'lsr', 'asr'
+    )),
+
+    _mult_operator: $ => token(choice(
+      seq(/[*/%]/, repeat(OP_CHAR)),
+      'mod', 'land', 'lor', 'lxor'
+    )),
+
+    _add_operator: $ => token(choice(
+      seq('+', repeat(OP_CHAR)),
+      seq('-', choice(optional(/[!$%&*+\-./:<=?@^|~]/), repeat2(OP_CHAR))),
+    )),
+
+    _concat_operator: $ => token(
+      seq(/[@^]/, repeat(OP_CHAR))
+    ),
+
+    _rel_operator: $ => token(choice(
+      seq(/[=>$]/, repeat(OP_CHAR)),
+      seq('<', choice(optional(/[!$%&*+./:<=>?@^|~]/), repeat2(OP_CHAR))),
+      seq('&', choice(/[!$%*+\-./:<=>?@^|~]/, repeat2(OP_CHAR))),
+      seq('|', choice(/[!$%&*+\-./:<=>?@^~]/, repeat2(OP_CHAR))),
+      '!='
+    )),
+
+    _and_operator: $ => /&&?/,
+
+    _or_operator: $ => choice('or', '||'),
+
+    _assign_operator: $ => choice('<-', ':='),
+
+    dot_operator: $ => token(
+      seq(/[!$%&*+\-/:=>?@^|~]/, repeat(OP_CHAR))
+    ),
+
+    dot_operator_path: $ => choice(
+      $.dot_operator,
+      seq($.module_name, '.', $.dot_operator_path)
+    ),
+
+    value_name: $ => choice(
+      LOWERCASE_IDENT,
+      parenthesize(choice(
+        $.prefix_operator,
+        alias($._sign_operator, $.infix_operator),
+        $.infix_operator,
+        seq('.', $.dot_operator, '(', ')', optional('<-'))
+      ))
+    ),
+
+    value_path: $ => choice(
+      $.value_name,
+      seq($.module_name, '.', $.value_path)
+    ),
+
+    module_name: $ => prec(1, CAPITALIZED_IDENT),
+
+    module_type_name: $ => choice(CAPITALIZED_IDENT, LOWERCASE_IDENT),
+
+    module_type_path: $ => choice(
+      $.module_type_name,
+      seq($._module_expression, '.', $.module_type_name)
+    ),
+
+    _label_name: $ => alias(LOWERCASE_IDENT, $.label),
+
+    label: $ => token(seq(
+      /[?~]/,
+      LOWERCASE_IDENT
+    )),
+
+    _label_colon: $ => token(seq(
+      /[?~]/,
+      LOWERCASE_IDENT,
+      ':'
+    )),
+
+    field_name: $ => LOWERCASE_IDENT,
+
+    field_path: $ => choice(
+      $.field_name,
+      seq($.module_name, '.', $.field_path)
+    ),
+
+    constructor_name: $ => CAPITALIZED_IDENT,
+
+    constructor_path: $ => prec.right(choice(
+      $.constructor_name,
+      seq($.module_name, '.', $.constructor_path)
+    )),
+
+    type_constructor: $ => LOWERCASE_IDENT,
+
+    _extended_module_name: $ => seq(
+      $.module_name,
+      repeat(parenthesize($.module_path))
+    ),
+
+    module_path: $ => choice(
+      $._extended_module_name,
+      seq($._extended_module_name, '.', $.module_path)
+    ),
+
+    type_constructor_path: $ => choice(
+      $.type_constructor,
+      seq($._extended_module_name, '.', $.type_constructor_path)
+    ),
+
+    type_variable: $ => token(seq("'", IDENT)),
+
+    tag: $ => token(seq('`', IDENT)),
+
+    attribute_id: $ => sep1('.', IDENT)
   },
 
+  conflicts: $ => [
+    [$._module_type, $._module_expression]
```

Comment:
This conflict is caused by `$.include_statement`, which supports both modules (when used in `.ml` files or structures) and module types (when used in `.mli` files or signatures).